src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java

```diff
@@ -0,0 +1,49 @@
+package org.fogbowcloud.saps.engine.core.catalog;
+
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.List;
+
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.CatalogException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.TaskNotFoundException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.UserNotFoundException;
+import org.fogbowcloud.saps.engine.core.dispatcher.notifier.Ward;
+import org.fogbowcloud.saps.engine.core.model.SapsImage;
+import org.fogbowcloud.saps.engine.core.model.SapsUser;
+import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
+
+public interface Catalog {
+
+    SapsImage addTask(String taskId, String dataset, String region, Date date, int priority, String user,
+                      String inputdownloadingPhaseTag, String digestInputdownloading, String preprocessingPhaseTag,
+                      String digestPreprocessing, String processingPhaseTag, String digestProcessing) throws CatalogException;
+
```

Comment:
this method is complex to understand. doc it. can we have multiple changetimestamps per state?

Reply:
I mean, "add" means we can have multiple

Reply:
At the moment, for each state there is only one changetimestamp, but this is done through removals and additions instead of updates.